src/sentry/services/hybrid_cloud/users.py

```diff
@@ -0,0 +1,82 @@
+from abc import abstractmethod
+from typing import Iterable, List, Optional, Sequence
+
+from sentry.models.group import Group
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.user import User
+from sentry.services.hybrid_cloud import (
+    CreateStubFromBase,
+    InterfaceWithLifecycle,
+    silo_mode_delegation,
+)
+from sentry.silo import SiloMode
+
+
+class UserService(InterfaceWithLifecycle):
+    @abstractmethod
+    def get_users_by_email(self, email: str, is_active=True) -> List[User]:
+        """
+        Return a list of active users with verified emails matching the parameter
+        :param email:
+        A case insensitive email to match
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def get_from_group(self, group: Group) -> List[User]:
+        """Get all users in all teams in a given Group's project."""
+        pass
+
+    @abstractmethod
+    def get_users(self, user_ids: Iterable[int], is_active=True) -> List[User]:
+        """
+        This method returns User objects given an iterable of IDs
+        :param user_ids:
+        A list of user IDs to fetch
+        :return:
+        """
+        pass
+
+    def get_user(self, user_id: int, is_active=True) -> Optional[User]:
+        users = self.get_users([user_id], is_active=is_active)
+        if len(users) > 0:
+            return users[0]
+        else:
+            return None
+
+
+class DatabaseBackedUserService(UserService):
```

Comment:
The approach I took with organization, and membership, was to create new data classes and not use the existing ORM objects.  My thinking was that doing so would help with
1. Clarifying that these objects generally should not have further ORM object calls, such as on relations, or accidental saves.
2. Explicitly stating which fields were actually needed in the cross silo data case, so that when an api or alternative implementation was actually made, there was stronger clarity on which fields would be needed.

Do you imagine that a future pass at this breaks out an `ApiUser` data class object instead of the ORM?  It has implications across the codebase but I feel that it brings a lot of clarity with the other such objects.

Reply:
That makes total sense. I hadn't gotten that far, to be honest. My plan was to identify all the interface points to extract these interfaces, and then come back later to revisit the actual transport layer. Depending on how we implement that, it seemed sort of likely that there might be automated data classes generated from that work.

Reply:
Yup, that makes fair sense.   As long as we can come back around and lock that stuff down, extracting atleast the interface points is important.